From: Thomas Gleixner <tglx@linutronix.de>

setup_pit_timer is declared in asm-i386/timer.h.  Move it to the pit header
file, so it can be used by x86_64 as well.

Move also the PIT constants.

Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
Cc: Andi Kleen <ak@suse.de>
Cc: Ingo Molnar <mingo@elte.hu>
Cc: john stultz <johnstul@us.ibm.com>
Cc: Andi Kleen <ak@suse.de>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 arch/i386/kernel/i8253.c                 |    2 --
 arch/i386/kernel/vmiclock.c              |    1 +
 include/asm-i386/i8253.h                 |    7 +++++++
 include/asm-i386/mach-default/io_ports.h |    5 -----
 include/asm-i386/timer.h                 |    1 -
 5 files changed, 8 insertions(+), 8 deletions(-)

Index: linux-rt.q/arch/i386/kernel/i8253.c
===================================================================
--- linux-rt.q.orig/arch/i386/kernel/i8253.c
+++ linux-rt.q/arch/i386/kernel/i8253.c
@@ -14,8 +14,6 @@
 #include <asm/i8253.h>
 #include <asm/io.h>
 
-#include "io_ports.h"
-
 DEFINE_SPINLOCK(i8253_lock);
 EXPORT_SYMBOL(i8253_lock);
 
Index: linux-rt.q/arch/i386/kernel/vmiclock.c
===================================================================
--- linux-rt.q.orig/arch/i386/kernel/vmiclock.c
+++ linux-rt.q/arch/i386/kernel/vmiclock.c
@@ -32,6 +32,7 @@
 #include <asm/apicdef.h>
 #include <asm/apic.h>
 #include <asm/timer.h>
+#include <asm/i8253.h>
 
 #include <irq_vectors.h>
 #include "io_ports.h"
Index: linux-rt.q/include/asm-i386/i8253.h
===================================================================
--- linux-rt.q.orig/include/asm-i386/i8253.h
+++ linux-rt.q/include/asm-i386/i8253.h
@@ -3,6 +3,11 @@
 
 #include <linux/clockchips.h>
 
+/* i8253A PIT registers */
+#define PIT_MODE		0x43
+#define PIT_CH0			0x40
+#define PIT_CH2			0x42
+
 extern spinlock_t i8253_lock;
 
 extern struct clock_event_device *global_clock_event;
@@ -18,4 +23,6 @@ static inline void pit_interrupt_hook(vo
 	global_clock_event->event_handler(global_clock_event);
 }
 
+extern void setup_pit_timer(void);
+
 #endif	/* __ASM_I8253_H__ */
Index: linux-rt.q/include/asm-i386/mach-default/io_ports.h
===================================================================
--- linux-rt.q.orig/include/asm-i386/mach-default/io_ports.h
+++ linux-rt.q/include/asm-i386/mach-default/io_ports.h
@@ -7,11 +7,6 @@
 #ifndef _MACH_IO_PORTS_H
 #define _MACH_IO_PORTS_H
 
-/* i8253A PIT registers */
-#define PIT_MODE		0x43
-#define PIT_CH0			0x40
-#define PIT_CH2			0x42
-
 /* i8259A PIC registers */
 #define PIC_MASTER_CMD		0x20
 #define PIC_MASTER_IMR		0x21
Index: linux-rt.q/include/asm-i386/timer.h
===================================================================
--- linux-rt.q.orig/include/asm-i386/timer.h
+++ linux-rt.q/include/asm-i386/timer.h
@@ -5,7 +5,6 @@
 
 #define TICK_SIZE (tick_nsec / 1000)
 
-void setup_pit_timer(void);
 unsigned long long native_sched_clock(void);
 unsigned long native_calculate_cpu_khz(void);
 
